src/modules/SX127x.cpp

```diff
@@ -327,7 +363,31 @@ int16_t SX127x::receive(uint8_t* data, size_t len) {
       delete[] data;
       data = new uint8_t[length + 1];
     }
-    _mod->SPIreadRegisterBurst(SX127X_REG_FIFO, length, data);
+    else{
+      while(rcvByteCount < length){
```

Comment:
That one was my brain interpreting the inital part as "here's an empty buffer since you did not provide a buffer of appropriate size." Totally my mistake.